cmd/jaeger/internal/extension/jaegerstorage/extension.go

```diff
@@ -74,7 +75,15 @@ func GetMetricStorageFactory(name string, host component.Host) (storage.MetricSt
 	return mf, nil
 }
 
-func GetStorageFactoryV2(name string, host component.Host) (tracestore.Factory, error) {
+func GetTraceStoreFactory(name string, host component.Host) (tracestore.Factory, error) {
+	return getV2FactoryAdapter(name, host)
+}
+
+func GetDependencyStoreFactory(name string, host component.Host) (depstore.Factory, error) {
```

Comment:
I've been looking at missing test coverage and realized that it's hard to get error coverage for this code path. We have this Extension API:
```
type Extension interface {
	extension.Extension
	TraceStorageFactory(name string) (storage.Factory, bool)
	MetricStorageFactory(name string) (storage.MetricStoreFactory, bool)
}
```
Shouldn't it be exposing v2 factory instead of v1? And either it should have deps-related method OR there shouldn't be a static function for deps and instead it should be a runtime interface check, as we discussed on the ticket.

Maybe this is transitional? I actually don't like that the adapter is being applied by the static functions - I think the right way would be to have the extension itself expose v2 API (instead of v1 API - we could get to v1 via downgreading, if needed).